code/_onclick/ai.dm

```diff
@@ -63,6 +66,10 @@
 		waypoint_mode = 0
 		return
 
+	if(controlled_mech) //Are we piloting a mech? Placed here so the modifiers are not overridden.
```

Comment:
in python you could do something like this:

    def click_on_while_in_mech(a, params):
        '''implementation'''

    def click_on_not_in_mech(a, params):
        '''implementation'''

    def enter_mech(args):
        '''implementation'''
        ai.clickOn = click_on_while_in_mech
        '''the above line replaces ai.clickOn function'''

    def exit_mech(args):
        '''implementation'''
        ai.clickOn = click_on_not_in_mech

This effectively lets you skip checking if it's in a mech every time you click. I don't know if byond supports this.